src/evo/simplifiedmns.cpp

```diff
@@ -119,25 +119,27 @@ bool BuildSimplifiedMNListDiff(const uint256& baseBlockHash, const uint256& bloc
     AssertLockHeld(cs_main);
     mnListDiffRet = CSimplifiedMNListDiff();
 
-    BlockMap::iterator baseBlockIt = mapBlockIndex.begin();
+    const CBlockIndex* baseBlockIndex = chainActive[0];
```

Comment:
Why do we even start with genesis as default when `baseBlockHash` is "null"? It would probably make sense to bail out instead.

Reply:
That's because `protx diff 0000000000000000000000000000000000000000000000000000000000000000 <someRealHash>` is defined to return the full list, which is practically the same as returning the diff between genesis and `someRealHash`

Reply:
Ok, makes sense, at least this matches dip4. One small suggestion:

```suggestion
    const CBlockIndex* baseBlockIndex = chainActive.Genesis();
```